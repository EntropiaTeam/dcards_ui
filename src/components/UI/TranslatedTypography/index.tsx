import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { Typography, TypographyProps } from '@material-ui/core';

interface TranslatedTypographyProps extends TypographyProps {
  i18nKey: string;
}

const TranslatedTypography: FC<TranslatedTypographyProps> = ({
  i18nKey,
  variant = 'body1',
  ...props
}) => {
  const { t } = useTranslation();
  return <Typography variant={variant} {...props}>{t(i18nKey)}</Typography>;
};

export default TranslatedTypography;
