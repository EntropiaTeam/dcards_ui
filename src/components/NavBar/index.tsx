import React, {
  memo, FC, useCallback, ReactNode
} from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { AppBar, Button, Toolbar } from '@material-ui/core';
import { BackIcon, CloseIcon } from '../UI/Icons';
import TranslatedTypography from '../UI/TranslatedTypography';
import ProgressBar from '../UI/ProgressBar';
import { useNavigationContext } from '../../context';
import IframeActions from '../../utils/IframeActions';
import AdobeAnalytics from '../../utils/AdobeAnalytics';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    height: '84px'
  },
  navBtn: {
    flex: 1,
    minWidth: 'auto',
    padding: 0,
    [theme.breakpoints.only('xs')]: {
      minWidth: '64px'
    }
  },
  backIcon: {
    [theme.breakpoints.only('xs')]: {
      marginRight: 'auto'
    }
  },
  closeIcon: {
    [theme.breakpoints.only('xs')]: {
      marginLeft: 'auto'
    }
  },
  appBar: {
    borderRadius: '32px 32px 0 0',
    backgroundColor: 'white',
    color: theme.palette.primary.dark,
    boxShadow: 'none'
  },
  toolBar: {
    display: 'flex',
    flex: 1,
    justifyContent: 'space-between',
    minHeight: '80px',
    [theme.breakpoints.only('xs')]: {
      minHeight: '60px'
    }
  },
  headerTitle: {
    flex: 12,
    backgroundColor: '#ffffff',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    textAlign: 'center',
    fontSize: '2rem',
    margin: '0 10px',
    [theme.breakpoints.only('xs')]: {
      fontSize: '1.5rem'
    },
    '&::first-letter': {
      textTransform: 'capitalize'
    }
  }
}));

interface NavbarProps {
  pageTitle: string;
  onBackClick?(): void;
  className?: string;
  position?: 'static' | 'fixed';
  setModalCallback?: (callback: () => void) => void;
  showModal?: boolean;
  rightControlContent?: ReactNode;
}

const NavBar: FC<NavbarProps> = ({
  className = '',
  position = 'static',
  pageTitle,
  onBackClick = () => { },
  setModalCallback,
  showModal = false,
  rightControlContent
}) => {
  const { state: { isBackButtonVisible } } = useNavigationContext();
  const classes = useStyles();

  const onCloseClick = useCallback(() => {
    AdobeAnalytics.getInstance().cancelButton();
    IframeActions.closeWindow();
  }, []);

  const handleCloseClick = (): void => {
    if (setModalCallback) {
      setModalCallback(() => onCloseClick);
      return;
    }
    onCloseClick();
  };

  const handleBackClick = (): void => {
    if (showModal && setModalCallback) {
      setModalCallback(() => onBackClick);
      return;
    }
    onBackClick();
  };

  const backIconTabIndex = isBackButtonVisible ? 1 : -1;
  const closeIconTabIndex = 2;

  const rightNavControl = rightControlContent || (
  <Button
    className={classes.navBtn}
    onClick={handleCloseClick}
    tabIndex={closeIconTabIndex}
    role="button"
  >
    <CloseIcon className={classes.closeIcon} />
  </Button>
  );

  return (
    <div className={className}>
      <AppBar position={position} className={classes.appBar}>
        <Toolbar variant="dense" className={classes.toolBar}>
          {
            isBackButtonVisible ? (
              <Button className={classes.navBtn} onClick={handleBackClick} role="button" tabIndex={backIconTabIndex}>
                <BackIcon className={classes.backIcon} />
              </Button>
            ) : (
              <Button disabled className={classes.navBtn} style={{ color: 'transparent' }}>
                <BackIcon />
              </Button>
            )
          }
          <TranslatedTypography
            i18nKey={pageTitle}
            variant="h3"
            classes={{ root: classes.headerTitle }}
          />
          {rightNavControl}
        </Toolbar>
        <ProgressBar />
      </AppBar>
    </div>
  );
};

export default memo(NavBar);
