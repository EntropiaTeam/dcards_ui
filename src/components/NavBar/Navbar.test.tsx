import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Button } from '@material-ui/core';
import TranslatedTypography from '../UI/TranslatedTypography';
import NavBar from '.';

interface IntlMock {
  useTranslation: () => void;
}

interface TranslationMock {
  t: () => void;
}

interface ReactRouterDom {
  useLocation: () => void;
  useRouteMatch: () => void;
}

jest.mock('react-i18next', (): IntlMock => ({
  useTranslation: (): TranslationMock => ({
    t: (key: string): string => key
  })
}));

jest.mock('react-router-dom', (): ReactRouterDom => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
    search: '',
    hash: '',
    state: null,
    key: '5nvxpbdafa'
  }),
  useRouteMatch: jest.fn().mockReturnValue(true)
}));

describe('<NavBar />', () => {
  const NavbarComponent = renderer.create(<NavBar pageTitle="" />).toJSON();
  const wrapper = mount(<NavBar pageTitle="" />);

  it('should render Navbar component correctly', () => {
    expect(NavbarComponent).toMatchSnapshot();
  });

  it('should render one TranslatedTypography component', () => {
    expect(wrapper.find(TranslatedTypography)).toHaveLength(1);
  });

  it('should render one Button component', () => {
    expect(wrapper.find(Button)).toHaveLength(2);
  });
});
