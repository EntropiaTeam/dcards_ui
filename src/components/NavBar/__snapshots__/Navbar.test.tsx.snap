// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render Navbar component correctly 1`] = `
<div
  className=""
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-5 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-dense makeStyles-toolBar-6 MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navBtn-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            className="makeStyles-backIcon-3"
            fill="none"
            height="30"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.9688 6L10 14.9062L17.9688 23.8125"
              stroke="currentColor"
              strokeWidth="2"
            />
          </svg>
        </span>
      </button>
      <h3
        className="MuiTypography-root makeStyles-headerTitle-7 MuiTypography-h3"
      >
        
      </h3>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navBtn-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={2}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            className="makeStyles-closeIcon-4"
            fill="none"
            height="30"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.03125 7.03125L22.9688 22.9688"
              stroke="currentColor"
              strokeWidth="2"
            />
            <path
              d="M7 22.9375L22.9375 7"
              stroke="currentColor"
              strokeWidth="2"
            />
          </svg>
        </span>
      </button>
    </div>
  </header>
</div>
`;
